cmd/kube-apiserver/app/server.go

```diff
@@ -208,7 +212,29 @@ func Run(s *options.APIServer) error {
 	if err != nil {
 		glog.Errorf("Failed to create clientset: %v", err)
 	}
-	admissionController := admission.NewFromPlugins(client, admissionControlPluginNames, s.AdmissionControlConfigFile)
```

Comment:
Actually, it seems like you want exactly this, plus the ability to pass a `PluginIntializer`.

Alternatively, your `PluginInitializer` could have a `NewFromPlugins()` method on it, but I don't think that would allow the composition to run as cleanly with multiple possible initializers (kube and openshift as an example).  You could try it and see.